src/net/sourceforge/kolmafia/persistence/ConcoctionDatabase.java

```diff
@@ -1252,6 +1253,58 @@ public static final void refreshConcoctions(boolean force) {
     ConcoctionDatabase.refreshConcoctionsNow();
   }
 
+  /**
+   * Get first index of usableList that has sortOrder non-NONE, assuming the list is sorted by
+   * sortOrder.
+   */
+  private static OptionalInt getFirstIndexNotNone() {
+    int low = 0;
+    int high = ConcoctionDatabase.usableList.size();
+    while (low < high) {
+      int mid = (low + high) / 2;
+      if (ConcoctionDatabase.usableList.get(mid).sortOrder == Concoction.Priority.NONE) {
+        low = mid + 1;
+      } else {
+        high = mid;
+      }
+    }
+    if (ConcoctionDatabase.usableList.get(low).sortOrder == Concoction.Priority.NONE) {
+      return OptionalInt.empty();
+    } else {
+      return OptionalInt.of(low);
+    }
+  }
+
+  private static void fastSortUsableList() {
+    // Concoctions are fully sorted at initialization time. The first sorting parameter is the
```

Comment:
Good thought, thanks. The most common trigger for a database refresh is gaining or losing a single item in combat or elsewhere. That can do things like e.g. change the adventure gain for all foods because the item gained was a Special Seasoning. But most of the time it shouldn't change the sorting at all. We don't actually refresh inventory very often to my knowledge, so that's an infrequent use case. That's all to say that I think there is likely a way to reduce the number of database refreshes altogether, but I'd say it's out-of-scope for this PR.

Your other comment about the data structure is also helpful. I think it makes some sense, but would require a number of changes elsewhere in the code (`ConcoctionDatabase.getUsables().sort()` is a fairly common pattern). I'll see whether there is a good way to do that instead.

Reply:
> We don't actually refresh inventory very often to my knowledge

I checked this recently, because it's a synchronized method, so in real terms it's pretty expensive.

We refresh inventory:
* whenever we receive a Kmail (via notification in chat)
* whenever somebody attacks us (via notification in chat)

Plus some more reasonable cases:
* finishing the doctor bag quest when we don't know which item we had to collect
* visiting the manor without having a telegram, when we think we haven't started the quest
* ascending into Exploathing
* enter / leave limit mode
* KoL says we don't have something we thought we did
* pulling stuff from Hagnks
* manual refresh

So not that often unless we receive lots of kmails, but we can close chat to avoid that issue.

Reply:
After looking into broader restructuring for a little bit, it looks quite tricky. The kernel of the issue is that UseItemEnqueuePanel - the ultimate destination for the usableList - uses a different model of food/booze/spleen/potions (ConcoctionType) than the underlying list does. This is fine, since it just filters the sorted usableList and displays that, but breaking them out into separate food/booze/spleen lists based on Concoction.Priority and using those as the models for the various UseItemEnqueuePanels won't quite work without doing some tricky reconciliation between the two models. Ultimately, we should only be using one model of what is a food/booze/spleen/potion. I'll keep playing with it but may declare this out of scope.